File: assignments/assignment-2-solution/src-csharp/TeaShop/Domain/Payment/PaymentStrategyBase.cs
using Assignment2Solution.Domain.Inventory;

namespace Assignment2Solution.Domain.Payment;

/// <summary>
///     Base class for payment strategies that provides common functionality.
/// </summary>
public abstract class PaymentStrategyBase : IPaymentStrategy
{
    /// <inheritdoc />
    public abstract void Checkout(InventoryItem item, int quantity, TextWriter output);

    /// <summary>
    ///     Computes the total amount for a purchase.
    /// </summary>
    /// <param name="item">The inventory item.</param>
    /// <param name="quantity">The quantity.</param>
    /// <returns>The total price.</returns>
    protected decimal ComputeTotalAmount(InventoryItem item, int quantity)
    {
        return item.Price * quantity;
    }
}

File: assignments/assignment-2-solution/src-java/src/main/java/assignment2solution/domain/payment/PaymentStrategyBase.java
package assignment2solution.domain.payment;

import assignment2solution.domain.inventory.InventoryItem;

import java.math.BigDecimal;

/**
 * Base class for payment strategies that provides common functionality.
 */
public abstract class PaymentStrategyBase implements IPaymentStrategy {
    /**
     * Computes the total amount for a purchase.
     */
    protected BigDecimal computeTotalAmount(InventoryItem item, int quantity) {
        return item.getPrice().multiply(BigDecimal.valueOf(quantity));
    }
}
